src/org/omegat/util/StringUtil.java

```diff
@@ -511,4 +511,31 @@ public static String format(String str, Object... arguments) {
         str = str.replaceAll("'", "''");
         return MessageFormat.format(str, arguments);
     }
+
+    /**
+     * convert Zenkaku/Hankaku alphabet to regex  that match both
+     *
+     * replace (\uFF01 - \uFF5E) to (\u0021- \u007E)
+     * such as text = text.replaceAll("u\0021", "(\uFF01|\u0021)");
+     * except for chars that consist of regex.
+     * FIXME: handling 0x20-0x2f, 0x5c
+     *
+     * @param text
+     * @return String with regex
+     */
+    public static String replaceZenHan(String text) {
+        int i;
+        StringBuilder result = new StringBuilder(text.length());
+        for (i = 0; i < text.length(); i++ ) {
+            char c = text.charAt(i);
+            if (0x0030 < (int) c && (int) c  < 0x007f && (int) c != 0x5c) {
+                result.append(String.format("(\\u%04x|\\u%04x)", (int) c - 0x0021 + 0xff01, (int) c ));
```

Comment:
Just FYI: `String.format` is very slow in Java. Since this method is going to be used in a performance-sensitive situation, I would rather see `Integer.toHexString` and multiple `appends` to `result`.